client/src/utils/ajax.js

```diff
@@ -1,41 +1,48 @@
 import envData from '../../../config/env.json';
-import axios from 'axios';
 import Tokens from 'csrf';
 import cookies from 'browser-cookies';
 
-const { apiLocation } = envData;
+const { apiLocation, environment } = envData;
 
 const base = apiLocation;
 const tokens = new Tokens();
 
-axios.defaults.withCredentials = true;
+// TODO: test on staging.  Do we need 'include' everywhere?
+const defaultOptions = {
+  credentials: environment === 'development' ? 'include' : 'same-site'
+};
 
 // _csrf is passed to the client as a cookie. Tokens are sent back to the server
 // via headers:
-function setCSRFTokens() {
+function getCSRFToken() {
   const _csrf = typeof window !== 'undefined' && cookies.get('_csrf');
-  if (!_csrf) return;
-  axios.defaults.headers.post['CSRF-Token'] = tokens.create(_csrf);
-  axios.defaults.headers.put['CSRF-Token'] = tokens.create(_csrf);
+  return _csrf ? tokens.create(_csrf) : '';
 }
 
 function get(path) {
-  return axios.get(`${base}${path}`);
+  return fetch(`${base}${path}`, defaultOptions).then(res => res.json());
 }
 
 export function post(path, body) {
-  setCSRFTokens();
-  return axios.post(`${base}${path}`, body);
+  return request('POST', path, body);
 }
 
 function put(path, body) {
-  setCSRFTokens();
-  return axios.put(`${base}${path}`, body);
+  return request('PUT', path, body);
 }
 
-// function del(path) {
-//   return axios.delete(`${base}${path}`);
-// }
+function request(method, path, body) {
+  const options = {
+    ...defaultOptions,
+    method,
+    headers: {
+      'CSRF-Token': getCSRFToken(),
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  };
+  return fetch(`${base}${path}`, options).then(res => res.json());
```

Comment:
Okay this took a bit to debug, but...

Certain endpoints we use don't actually send back JSON, such as the `/account/reset-progress` endpoint. When attempting to perform this action, the `res.json()` call throws an error and we get the "Something went wrong" flash message.

Reply:
For clarification, this was the response and result of res.json():

![image](https://user-images.githubusercontent.com/63889819/117363259-30e84d00-ae71-11eb-9cab-2e0bfe66e33f.png)

For testing, I changed it to `res.text()`:

![image](https://user-images.githubusercontent.com/63889819/117363291-3d6ca580-ae71-11eb-8b84-bc12e5b4a701.png)

Reply:
@ojeytonwilliams what's the performance increase here? Is it worth the overhead of potentially refactoring our API to always send JSON?

Reply:
Nice job catching and debugging this.  I need to be more careful in future.

The decrease in the bundle is fairly modest.  However I think unifying how the API responds is worthwhile in and of itself, so I'll take a look.  Before I do any of that, though, I'm going to add some more tests!